api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,21 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  try {
+    const token = req.headers['x-access-token'];
+    if (!token) throw new ClientError(401, 'authentication required');
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    console.error(err);
```

Comment:
Here you need to call `next` with the `err`, otherwise if there is an error your server would just hang here.